### DISCUSSION TITLE
Could anybody explain for what reason I have created the following filter?

### DISCUSSION THREAD

**Discussion by johnmapeson** [jgm/pandoc#10050] [Q&A]
This is kinda a weird question.

In April I asked how to create a Lua filter to preserve videos if converting from HTML to GitHub Flavored Markdown. @jgm helped me, and here are a link: https://github.com/jgm/pandoc/discussions/9658 and code:

``` sh
pandoc -f html+raw_html input.htm -t gfm-raw_html -o output.md --wrap=none --lua-filter preserve-videos.lua
```

``` lua
local function starts_with(start, str)
  return str:sub(1, #start) == start
end

function RawBlock (el)
  return ((starts_with('<video', el.text)
      or starts_with('</video', el.text)
      or starts_with('<source', el.text))
      and pandoc.RawBlock('markdown', el.text))
      or {}
end
```

But what I don't understand now is that for what reason I have created this filter. Isn't it 

``` sh
pandoc -f html+raw_html input.htm -t gfm-raw_html -o output.md --wrap=none
```

preserves videos even without it?

I suspect the filter does have its use, but I forgot the use that does make sense for it. Any help is *much* appreciated!

**Comment by alerque** [comment]
Given that your original example is not currently reproducible (I just used Pandoc 3.3 to run your input and command and got a different result) I would guess the answer is that something got fixed/changed in the mean time and the need for the filter is obsolete.

**Comment by johnmapeson** [comment]
Hello, Caleb. I cannot test it with Pandoc 3.3, but I have just updated Pandoc to 2.3.1 by using `brew upgrade`, and here is how it works for me:

`input.htm`:

```
<!-- comment -->
<h1>heading</h1>
<p>body</p>
<img src="image.png"</img>
<p>body</p>
<video controls><source src="video.mp4"></video>
<p>body</p>
```

Then, either...

- `pandoc -f html+raw_html input.htm -t gfm-raw_html -o output.md --lua-filter preserve-videos.lua`
- `pandoc -f html+raw_html input.htm -t gfm-raw_html -o output.md`

produces the same `output.md`:

```
# heading

body

![](image.png)

body

<video controls>

<source src="video.mp4">

</video>

body
```

The output files are the same and the video is preserved even without a filter.
